File: src/AceRental.Application/Clients/Dtos/ClientDto.cs
using AceRental.Application.Reservations.Dtos;

namespace AceRental.Application.Clients.Dtos
{
    public class ClientDto
    {
        public Guid Id { get; set; }
        public required string ClientNumber { get; set; }
        public string? RaisonSociale { get; set; }
        public required string FirstName { get; set; }
        public required string LastName { get; set; }
        public required string Email { get; set; }
        public string? PhoneNumber { get; set; }
        public string? TelNumber { get; set; }
        public string? Address { get; set; }
        public string? ComplementAdresse { get; set; }
        public string? City { get; set; }
        public int PostalCode { get; set; }
        public ICollection<ReservationDto> Reservations { get; set; } = new List<ReservationDto>();
    }
}

File: AceRental.FrontEnd/ClientApp/src/app/interfaces/client-dto.ts
export interface ClientDto {
    Id?: string;
    ClientNumber: string;
    RaisonSociale?: string;
    LastName : string;
    FirstName : string;
    Email : string;
    Password: string;
    Civilite: string;
    TelNumber?: string;
    PhoneNumber?: string;
    Address?: string;
    ComplementAdresse?: string;
    PostalCode: number;
    City?: string;
    CreateAt: Date;
}
